src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -796,19 +806,19 @@ public virtual Task<TUser> FindByLoginAsync(string loginProvider, string provide
         {
             ThrowIfDisposed();
             var loginStore = GetLoginStore();
-            if (loginProvider == null)
+            if (user == null)
```

Comment:
nit: lets not change the ordering of these checks, its possible the unit tests rely on the current priority